core/src/main/java/com/linecorp/armeria/common/util/Sampler.java

```diff
@@ -42,6 +42,64 @@
  */
 @FunctionalInterface
 public interface Sampler<T> {
+
+    /**
+     * Returns a sampler that applies logical or operator to both samplers decisions.
+     */
+    default Sampler<T> or(Sampler<T> other) {
+        return new OrSampler<>(this, other);
```

Comment:
```suggestion
        return new OrSampler<>(this, requireNonNull(other, "other"));
```
Can we also consider the situation where more than two samplers are chained? e.g. `a.or(b).or(c)` by accepting an array of samplers in `OrSampler.<init>()`?

Reply:
If we accept an array of samplers, a new syntax can be used `a.or(b, c, d)`. Is this syntax  something we want?

Because `.or` returns another sampler, we can safely call another `.or` as you described.

`a.or(b).or(c)` looks better than `a.or(b, c)` in my opinion, what do you think?

Reply:
Ah, I was actually talking about the constructor of `OrSampler()`. `or()` should accept only one parameter. `OrSampler.or()` could be optimized so that `OrSampler` doesn't wrap another `OrSampler`.